nanomq/pub_handler.c

```diff
@@ -921,6 +917,9 @@ rule_engine_insert_sql(nano_work *work)
 				snprintf(key, 128, "%s (", rules[i].sqlite_table);
 				char value[800]       = "VALUES (";
 				for (size_t j = 0; j < 9; j++) {
+					if (rule_mutex == NULL) {
+						nng_mtx_alloc(&rule_mutex);
```

Comment:
Such an expensive insert. Even need to allocate a mutex every time.

Reply:
And this will cause data race error in rule_mutex itself.
These changes would also make the mutex loses the role it's supposed to play.

Example.
In A thread. Rule_mutex is allocated and was locked. In B thread. Because of exists of rule_mutex. It pass the process of allocation. And wait to lock rule_mutex; After A thread unlock the rule_mutex and free the rule_mutex. B thread get a null rule_mutex.